create.go

```diff
@@ -255,46 +260,41 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, obj interface{
 			k8sClient = k8sClients.K8sClient()
 		}
 
-		if k8sClient != nil {
```

Comment:
There would be no case k8sclient will be `nil`, it will always return out of function before then.